core/src/avm1/function.rs

```diff
@@ -315,13 +315,18 @@ impl<'gc> Executable<'gc> {
                 };
 
                 let max_recursion_depth = activation.context.avm1.max_recursion_depth();
+                let base_clip = if effective_ver > 5 {
+                    af.base_clip
+                } else {
+                    this.as_display_object().unwrap()
```

Comment:
This unwrap panics on an SWF5 swf:
```

var o = {
	f: function() {
		trace("f");
		this.g();
	},
	g: function() {
		trace("g");
	}
};
o.f();
```

Reply:
Tested this (`af.base_clip.avm1_root().unwrap()`) but this is causing a regression in [Spike Boy](https://www.y8.com/games/spike_boy) (by the way, the game also needs #2636): pressing the arrow keys moves the GUI instead of moving the character.

Reply:
It now fallbacks to `activation.base_clip()`.
I think that `af.base_clip.avm1_root()` (and `af.base_clip`) are incorrect because [test.zip](https://github.com/ruffle-rs/ruffle/files/5852172/test.zip) shows a different behavior.